xarray/core/dataset.py

```diff
@@ -509,14 +509,23 @@ def __dask_graph__(self):
         if not graphs:
             return None
         else:
-            from dask import sharedict
-            return sharedict.merge(*graphs.values())
+            try:
+                from dask.highlevelgraph import HighLevelGraph
+                return HighLevelGraph.merge(*graphs.values())
+            except ImportError:
+                from dask import sharedict
+                return sharedict.merge(*graphs.values())
+
 
     def __dask_keys__(self):
         import dask
         return [v.__dask_keys__() for v in self.variables.values()
                 if dask.is_dask_collection(v)]
 
+    def __dask_layers__(self):
+        return sum([v.__dask_layers__() for v in self.variables.values() if
+                    dask.is_dask_collection(v)], [])
```

Comment:
I think you need to add a local `import dask` here since dask isn't a hard dependency. This suggests there isn't any coverage for this new method -- is there any easy way to add that?